isis/src/base/objs/TrackingTable/TrackingTable.cpp

```diff
@@ -0,0 +1,165 @@
+/**
+ * @file
+ * $Revision: 1.4 $
+ * $Date: 2010/05/14 19:17:09 $
+ *
+ *   Unless noted otherwise, the portions of Isis written by the USGS are public
+ *   domain. See individual third-party library and package descriptions for
+ *   intellectual property information,user agreements, and related information.
+ *
+ *   Although Isis has been used by the USGS, no warranty, expressed or implied,
+ *   is made by the USGS as to the accuracy and functioning of such software
+ *   and related material nor shall the fact of distribution constitute any such
+ *   warranty, and no responsibility is assumed by the USGS in connection
+ *   therewith.
+ *
+ *   For additional information, launch
+ *   $ISISROOT/doc//documents/Disclaimers/Disclaimers.html in a browser or see
+ *   the Privacy &amp; Disclaimers page on the Isis website,
+ *   http://isis.astrogeology.usgs.gov, and the USGS privacy and disclaimers on
+ *   http://www.usgs.gov/privacy.html.
+ */
+
+#include "TrackingTable.h"
+
+#include <QList>
+#include <QString>
+
+#include "FileName.h"
+#include "IException.h"
+#include "Pvl.h"
+#include "Table.h"
+#include "TableField.h"
+#include "TableRecord.h"
+
+using namespace std;
+namespace Isis {
+
+
+  /**
+  * @brief Default constructor
+  */
+  TrackingTable::TrackingTable() {
+    
+  }
+  
+  
+  /**
+  * @ brief Constructs a TrackingTable given a Table object. The Table is used to populate 
+  *         m_fileList.
+  *
+  * @param table The Table object to pull the filenames and serial numbers from
+  */
+  TrackingTable::TrackingTable(Table table) {
+    
+    for (int i=0; i < table.Records(); i++) {
+      TableRecord record = table[i];
+      QString nameField = QString(record["FileName"]).toLatin1().data();
```

Comment:
We don't need the toLatin1 and data calls here, they convert a QString into a Char*. Then, the QString assignment operator converts it back to a QString.